nilearn/regions/region_extractor.py

```diff
@@ -0,0 +1,320 @@
+            mask_img = resample_img(mask_img,
+                                    target_affine=maps_img.get_affine(),
+                                    target_shape=maps_img.shape[:3],
+                                    interpolation="nearest")
+            mask_data, _ = masking._load_mask_img(mask_img)
+            # Set as 0 to the values which are outside of the mask
+            maps[mask_data == 0.] = 0.
+
+    for index in range(maps.shape[-1]):
+        regions = []
+        map_ = maps[..., index]
+        # Mark the seeds using random walker
+        if extract_type == 'local_regions':
+            smooth_map = _smooth_array(map_, affine, peak_local_smooth)
+            seeds = _peak_local_max(smooth_map, indices=False)
+            seeds_label, seeds_id = label(seeds)
+            # Assign -1 to values which are 0. to indicate to ignore
+            seeds_label[map_ == 0.] = -1
+            rw_maps = _random_walker(map_, seeds_label, mode='cg')
+            # Now simply replace "-1" with "0" for regions seperation
+            rw_maps[rw_maps == -1] = 0.
+            label_maps = rw_maps
+        else:
+            # Taking directly the input data which is the case in 'auto' type
+            maps_assigned = map_
+            # Connected component extraction
+            label_maps, n_labels = label(maps_assigned)
+
+        # Takes the size of each labelized region data
+        labels_size = np.bincount(label_maps.ravel())
+        labels_size[0] = 0.
+        for label_id, label_size in enumerate(labels_size):
+            if label_size > min_size:
+                region_data = (label_maps == label_id) * map_
+                region_img = new_img_like(maps_img, region_data)
+                regions.append(region_img)
+
+        index_of_each_map.extend([index] * len(regions))
+        all_regions_imgs.extend(regions)
+
+    regions_extracted = concat_niimgs(all_regions_imgs)
+
+    return regions_extracted, index_of_each_map
+
+
+class RegionExtractor(NiftiMapsMasker):
+    """ Class to extract connected regions into seperate regions.
+
+    Region Extraction is a post processing technique which
+    is implemented to automatically segment each brain atlas maps
+    into different set of seperated brain activated region.
+    Particularly, to show that each decomposed brain maps can be
+    used to focus on a target specific Regions of Interest analysis.
+
+    Parameters
+    ----------
+    maps_img: 4D Niimg-like object or path to the 4D Niimg
+       an image or a filename of the image which contains a set of whole brain
+       atlas maps or statistically decomposed brain maps.
+
+    mask_img: Niimg-like object or None,  default is None, optional
+        Mask to be applied on the input data.
+        If given, mask_img is submitted to an instance of NiftiMapsMasker
+        for a preparation towards input maps data.
+        If None, no mask will be applied on the data.
+
+    min_size: int, default size 50, optional
```

Comment:
Maybe min_region_voxels or min_region_size?

Also no need for "size" after default:
int, default 50, optional